api/src/main/java/io/strimzi/api/kafka/model/listener/v2/KafkaListenerType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener.v2;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum KafkaListenerType {
+    INTERNAL,
```

Comment:
https://www.baeldung.com/java-enum-values
We can set values to the enums to remove all of the boilerplate `toValue`, `forValue` i believe

Reply:
You mean to use e.g. `INTERNAL("internal")` instead of having the `forValue` / `toValue` methods? IIRC that does not work well with the (de)serialization in the APIs.

Reply:
Ah that makes sense, apologies didn't know that!